File: enterprise-terminal/src/main/java/com/cyx/enterpriseterminal/entity/po/Project.java
package com.cyx.enterpriseterminal.entity.po;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("biz_project")
public class Project implements Serializable {



    /**
     * 项目ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 项目编号
     */
    private String projectCode;

    /**
     * 项目名称
     */
    private String projectName;

    /**
     * 建设单位ID
     */
    private Long enterpriseId;

    /**
     * 监理单位ID
     */
    private Long supervisorOrgId;

    /**
     * 项目类型
     */
    private String projectType;

    /**
     * 项目状态
     */
    private String projectStatus;

    /**
     * 项目地址
     */
    private String address;

    /**
     * 经度
     */
    private BigDecimal longitude;

    /**
     * 纬度
     */
    private BigDecimal latitude;

    /**
     * 计划开始日期
     */
    private LocalDate plannedStartDate;

    /**
     * 计划结束日期
     */
    private LocalDate plannedEndDate;

    /**
     * 实际开始日期
     */
    private LocalDate actualStartDate;

    /**
     * 实际结束日期
     */
    private LocalDate actualEndDate;

    /**
     * 项目进度百分比
     */
    private BigDecimal progressPercent;

    /**
     * 项目负责人
     */
    private String projectManager;

    /**
     * 负责人手机号
     */
    private String managerMobile;

    /**
     * 项目描述
     */
    private String description;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;

    /**
     * 逻辑删除：0未删除，1已删除
     */
    @TableLogic(value = "0", delval = "1")
    private Integer deleted;
}

File: UI/src/api/project.ts
import { http, type ApiResult } from './http'

export interface ProjectRecord { id:number;projectCode:string;projectName:string;enterpriseId:number;enterpriseName?:string;supervisorOrgId:number;projectType:string;projectStatus:string;address?:string;longitude?:number;latitude?:number;plannedStartDate?:string;plannedEndDate?:string;actualStartDate?:string;progressPercent?:number;projectManager?:string;managerMobile?:string;description?:string;members?:Array<{id:number;projectId:number;userId:number;memberRole:string}> }
export type ProjectPayload = Omit<ProjectRecord,'id'|'enterpriseName'|'members'> & {id?:number}
export interface EnterpriseProjectDetail {
  id:number
  projectCode:string
  projectName:string
  enterpriseName?:string
  supervisorName?:string
  projectType:string
  projectStatus:string
  address?:string
  longitude?:number
  latitude?:number
  plannedStartDate?:string
  plannedEndDate?:string
  actualStartDate?:string
  actualEndDate?:string
  progressPercent?:number
  projectManager?:string
  projectManagerPhone?:string
  description?:string
  createTime?:string
}
export interface EnterpriseProjectUpdatePayload {
  id:number
  projectCode:string
  projectName:string
  enterpriseName?:string
  supervisorName?:string
  projectType:string
  projectStatus:string
  address?:string
  projectManager?:string
  progressPercent?:number
  projectManagerPhone?:string
  description?:string
}
export interface EnterpriseRecord {id:number;organizationId?:number;enterpriseName:string;unifiedSocialCreditCode?:string;legalRepresentative?:string;contactName?:string;contactMobile?:string;address?:string;qualificationInfo?:string;status?:number}

export const getProjectList=()=>http.get<never,ApiResult<ProjectRecord[]>>('/supervisor/projects')
export const getProjectDetail=(id:number)=>http.get<never,ApiResult<ProjectRecord>>(`/supervisor/projects/${id}`)
export const createProject=(data:ProjectPayload)=>http.post<never,ApiResult<number>>('/supervisor/projects',data)
export const updateProject=(data:ProjectPayload)=>http.put<never,ApiResult<void>>('/supervisor/projects',data)
export const deleteProject=(id:number)=>http.delete<never,ApiResult<void>>(`/supervisor/projects/${id}`)
/** 企业端项目详情、修改、删除，与 /enterprise/project 控制器逐字段对应。 */
export const getEnterpriseProjectDetail=(projectId:number)=>http.get<never,ApiResult<EnterpriseProjectDetail>>(`/enterprise/project/details/${projectId}`)
export const updateEnterpriseProject=(data:EnterpriseProjectUpdatePayload)=>http.put<never,ApiResult<boolean>>('/enterprise/project/update',data)
export const deleteEnterpriseProject=(projectId:number)=>http.delete<never,ApiResult<boolean>>(`/enterprise/project/delete/${projectId}`)
export const replaceProjectMembers=(id:number,members:Array<{userId:number;memberRole:string}>)=>http.put<never,ApiResult<void>>(`/supervisor/projects/${id}/members`,members)
export const getEnterpriseList=()=>http.get<never,ApiResult<EnterpriseRecord[]>>('/supervisor/enterprises')
export const getEnterpriseDetail=(id:number)=>http.get<never,ApiResult<EnterpriseRecord>>(`/supervisor/enterprises/${id}`)
export const createEnterprise=(data:Omit<EnterpriseRecord,'id'>)=>http.post<never,ApiResult<number>>('/supervisor/enterprises',data)
export const updateEnterprise=(data:EnterpriseRecord)=>http.put<never,ApiResult<void>>('/supervisor/enterprises',data)
export const deleteEnterprise=(id:number)=>http.delete<never,ApiResult<void>>(`/supervisor/enterprises/${id}`)
